### PR TITLE
Add macros.h

### PR DESCRIPTION
**Pull request by nomis52**

### CODE REVIEW

File: firmware/src/macros.h [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+#ifndef FIRMWARE_SRC_MACROS_H_
+#define FIRMWARE_SRC_MACROS_H_
+
+/**
+ * @def UNUSED
+ * @brief Mark unused arguments & types.
+ *
+ * @examplepara
+ *   @code
+ *   void Foo(OLA_UNUSED int bar) {}
```
Comment (peternewman) [inline firmware/src/macros.h]: Fix the example

File: firmware/src/macros.h [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+ *
+ *   OLA_UNUSED typedef int Baz;
```
Comment (peternewman) [inline firmware/src/macros.h]: Again

**Comment by peternewman** [comment]
LGTM when comments fixed

File: firmware/src/macros.h (hunk 2 above)
Comment (nomis52) [inline firmware/src/macros.h]: Done

**Pull request merged**